fsspec/tests/conftest.py

```diff
@@ -68,10 +78,12 @@ def do_GET(self):
         if file_path.endswith("/") and file_path.rstrip("/") in self.files:
             file_path = file_path.rstrip("/")
         file_data = self.files.get(file_path)
+        if callable(file_data):
+            file_data = file_data(self.server.server_port)
         if "give_path" in self.headers:
             return self._respond(200, data=json.dumps({"path": self.path}).encode())
         if "redirect" in self.headers and file_path != "/index/realfile":
-            new_url = f"http://127.0.0.1:{port}/index/realfile"
+            new_url = f"http://127.0.0.1:{self.server.server_port}/index/realfile"
```

Comment:
self.server.realfile?

Reply:
`self.server` is the `HTTPServer`, not the `server` fixture, and doesn't have a `realfile` attribute. But this could probably use `_make_realfile` instead.